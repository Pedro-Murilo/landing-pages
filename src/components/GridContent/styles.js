import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacing.sizes.huge} 0;
  `}
`;
