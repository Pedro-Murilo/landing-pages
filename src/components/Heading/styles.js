import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.xmedium};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.big};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, darkTheme, size, uppercase }) => css`
    color: ${darkTheme ? theme.colors.red : theme.colors.green};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
