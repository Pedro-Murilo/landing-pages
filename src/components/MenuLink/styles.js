import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacing.sizes.small};
    color: ${theme.colors.green};
    position: relative;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.red};
      transition: all 0.3s ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }
  `}
`;
