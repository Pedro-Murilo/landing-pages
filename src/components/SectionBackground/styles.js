import styled, { css } from 'styled-components';

const isBackgroundActive = (theme) => css`
  background: ${theme.colors.black};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.black};
    ${background && isBackgroundActive(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
