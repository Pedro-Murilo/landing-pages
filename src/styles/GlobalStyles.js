import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 62.5%;
  }

  body {
    font-size: 1.6rem;
    font-family: 'Open Sans', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat', sans-serif;
    margin: ${({ theme }) => theme.spacing.sizes.xmedium} 0;
  }

  p {
    margin: ${({ theme }) => theme.spacing.sizes.medium} 0;
  }

  ul, ol {
    margin: ${({ theme }) => theme.spacing.sizes.medium};
    padding: ${({ theme }) => theme.spacing.sizes.medium};
  }

  a {
    color: ${({ theme }) => theme.colors.red};
  }
`;

export default GlobalStyles;
