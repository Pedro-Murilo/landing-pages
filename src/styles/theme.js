export const theme = {
  colors: {
    black: '#252525',
    white: '#f7f5f5',
    green: '#104f21',
    red: '#e34040',
    mediumGray: '#DDD',
  },
  font: {
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      xmedium: '3.2rem',
      big: '4rem',
      xbig: '4.8rem',
      huge: '5.6rem',
    },
  },
  spacing: {
    sizes: {
      xsmall: '0.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      xmedium: '3.2rem',
      big: '4rem',
      xbig: '4.8rem',
      huge: '5.6rem',
    },
  },
  media: {
    lteMedium: '(max-width: 768px)',
  },
};
