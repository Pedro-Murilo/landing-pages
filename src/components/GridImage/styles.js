import styled, { css } from 'styled-components';

import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacing.sizes.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacing.sizes.xmedium};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 200ms ease-in;

    &:hover {
      transform: scale(1.1) rotate(5deg);
    }
  `}
`;
