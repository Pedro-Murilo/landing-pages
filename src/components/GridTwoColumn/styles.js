import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacing.sizes.medium};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacing.sizes.xmedium};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacing.sizes.medium};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css``}
`;

export const Image = styled.img`
  ${() => css``}
`;
