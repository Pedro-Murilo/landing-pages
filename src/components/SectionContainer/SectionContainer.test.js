import { renderTheme } from '../../styles/RenderTheme';
import { screen } from '@testing-library/react';

import { SectionContainer } from '.';

describe('<SectionContainer />', () => {
  it('should render content', () => {
    const { container } = renderTheme(
      <SectionContainer>
        <h1>children</h1>
      </SectionContainer>,
    );
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
