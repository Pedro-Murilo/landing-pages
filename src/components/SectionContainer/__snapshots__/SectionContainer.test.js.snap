// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer /> should render content 1`] = `
.c0 {
  max-width: 120rem;
  margin: 0 auto;
}

<div>
  <div
    class="c0"
  >
    <h1>
      children
    </h1>
  </div>
</div>
`;
